src/isar/state_machine/states/monitor.py

```diff
@@ -30,37 +39,54 @@ def start(self):
 
     def stop(self):
         self.iteration_counter = 0
+        if self.mission_status_thread:
+            self.mission_status_thread.join()
+        self.mission_status_thread = None
+        self.mission_status_output = MissionStatusThreadOutput()
 
     def _run(self):
         while True:
             if self.state_machine.should_stop():
                 self.state_machine.stop_mission()
-
-            if not self.state_machine.status.mission_in_progress:
                 next_state = States.Cancel
                 break
 
-            mission_status: MissionStatus = self.state_machine.robot.mission_status(
-                self.state_machine.status.current_mission_instance_id
-            )
-            self.state_machine.status.mission_status = mission_status
-            self._log_status(mission_status=mission_status)
-
-            if self._mission_finished(
-                mission_status=mission_status,
-                instance_id=self.state_machine.status.current_mission_instance_id,
-            ):
-                if isinstance(
-                    self.state_machine.status.current_mission_step, DriveToPose
-                ):
-                    next_state = States.Send
-                else:
-                    next_state = States.Collect
-                break
-
             if self.state_machine.should_send_status():
                 self.state_machine.send_status()
-            time.sleep(self.state_machine.sleep_time)
+
+            if not self.mission_status_thread:
+                self.mission_status_thread = threading.Thread(
+                    target=mission_status_thread,
+                    args=(
```

Comment:
Could we list which input each of these would be, for instance: mission_status_func=self.state_machine.robot.mission_status?